### ISSUE TITLE
Will DGL support tensorflow?

### ISSUE THREAD

**Issue by F36** [dmlc/dgl#422]
Now, we know DGL is based on mxnet ecosystem.However, as many people are used tensorflow, will DGL support tensorflow?

Maybe, it's extremely difficult to solve it, but i am just curious to ask for it.

Thanks.

**Comment by jermainewang** [comment]
Hi,

**We really really like to support Tensorflow**. Unfortunately, none of our team members are heavy Tensorflow users so we are actively seeking volunteers on this task. If you know anyone who are willing to help, do not hesitate to refer them to this issue for discussion. We could leave this issue open so more can see this.

The task is not really extremely difficult. As you've noticed, DGL is compatible with two frameworks: MXNet and Pytorch. We have isolated the framework-dependent interface so adding new framework support can be done without major change to the core system. There are two major tasks:
* Implement the backend interface defined [here](https://github.com/dmlc/dgl/tree/master/python/dgl/backend). They are mostly straight forward tensor operations.
* To allow DGL input and output Tensorflow tensor object. We need Tensorflow to support DLPack standard. I have raised an issue [here](https://github.com/tensorflow/tensorflow/issues/24453).

And that should be it to support eager mode in Tensorflow. Supporting symbol mode requires more changes to DGL, but we are looking at how to support MXNet's symbol mode and the experience could be applied there.
